pkg/bindings/test/pods_test.go

```diff
@@ -0,0 +1,207 @@
+
+		// Adding an alpine container to the existing pod
+		bt.RunTopContainer(nil, &trueFlag, &newpod)
+		response, err := pods.Inspect(connText, newpod)
+		Expect(err).To(BeNil())
+
+		err = pods.Pause(connText, newpod)
+		Expect(err).To(BeNil())
+
+		// Binding needs to be modified to ispect the pod state.
+		// Since we dont have a pod state we inspect the states of the containers within the pod.
+		response, err = pods.Inspect(connText, newpod)
+		for _, i := range response.Containers {
+			Expect(define.StringToContainerStatus(i.State)).
+				To(Equal(define.ContainerStatePaused))
+		}
+
+		err = pods.Unpause(connText, newpod)
+		Expect(err).To(BeNil())
+		response, err = pods.Inspect(connText, newpod)
+		for _, i := range response.Containers {
+			Expect(define.StringToContainerStatus(i.State)).
+				To(Equal(define.ContainerStateRunning))
+		}
+	})
+
+	It("start stop restart pod", func() {
+		// Start an invalid pod
+		err = pods.Start(connText, "dummyName")
+		Expect(err).ToNot(BeNil())
+		code, _ := bindings.CheckResponseCode(err)
+		Expect(code).To(BeNumerically("==", http.StatusNotFound))
+
+		// Stop an invalid pod
+		err = pods.Stop(connText, "dummyName", nil)
+		Expect(err).ToNot(BeNil())
+		code, _ = bindings.CheckResponseCode(err)
+		Expect(code).To(BeNumerically("==", http.StatusNotFound))
+
+		// Restart an invalid pod
+		err = pods.Restart(connText, "dummyName")
+		Expect(err).ToNot(BeNil())
+		code, _ = bindings.CheckResponseCode(err)
+		Expect(code).To(BeNumerically("==", http.StatusNotFound))
+
+		// Start a valid pod and inspect status of each container
+		err = pods.Start(connText, newpod)
+		Expect(err).To(BeNil())
+
+		response, err := pods.Inspect(connText, newpod)
+		for _, i := range response.Containers {
+			Expect(define.StringToContainerStatus(i.State)).
+				To(Equal(define.ContainerStateRunning))
+		}
+
+		// To be fixed Start a already running container
+		// err = pods.Start(connText, newpod)
+		// Expect(err).ToNot(BeNil())
+		// code, _ = bindings.CheckResponseCode(err)
+		// Expect(code).To(BeNumerically("==", http.StatusNotModified))
+
+		// Stop the running pods
+		err = pods.Stop(connText, newpod, nil)
+		Expect(err).To(BeNil())
+		response, _ = pods.Inspect(connText, newpod)
+		for _, i := range response.Containers {
+			Expect(define.StringToContainerStatus(i.State)).
+				To(Equal(define.ContainerStateStopped))
+		}
+
+		// Stop a already running pod
+		// err = pods.Stop(connText, newpod, nil)
+		// Expect(err).ToNot(BeNil())
+		// code, _ = bindings.CheckResponseCode(err)
+		// Expect(code).To(BeNumerically("==", http.StatusNotModified))
+
+		err = pods.Restart(connText, newpod)
+		Expect(err).To(BeNil())
+		response, _ = pods.Inspect(connText, newpod)
+		for _, i := range response.Containers {
+			Expect(define.StringToContainerStatus(i.State)).
+				To(Equal(define.ContainerStateRunning))
+		}
+
+	})
+
+	// Remove all stopped pods and their container to be implemented.
+	It("prune pod", func() {
```

Comment:
Is this not implemented yet?

Reply:
The api binding is in there. Test to validate the endpoint needs to be implemented.